controllers/utils/failuredomains.go

```diff
@@ -111,8 +111,12 @@ func (c *CloudClientImplementation) AsFailureDomainUser(fdSpec *infrav1.CloudSta
 			return ctrl.Result{}, errors.Wrapf(err, "getting ACSEndpoint secret with ref: %v", fdSpec.ACSEndpoint)
 		}
 
+		clientConfig := &corev1.ConfigMap{}
+		key = client.ObjectKey{Name: cloud.ClientConfigMapName, Namespace: cloud.ClientConfigMapNamespace}
+		_ = c.K8sClient.Get(c.RequestCtx, key, clientConfig)
```

Comment:
do we need to check the return value here? Is it some unchecked error?

Reply:
if Get() returns an error then clientConfig will be null
if Get() returns a nil then clientConfig will not be null
Both cases will be handled in NewClientFromK8sSecret
if the input clientConfig is null then the default TTL will be used
if the input clientConfig is not null then it tries to use the TTL value in the configMap